src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1933,13 +1962,12 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            bool isGlobalScript = parentKind == SyntaxKind.CompilationUnit && this.IsScript;
-            bool acceptStatement = isGlobalScript;
+            bool isGlobal = parentKind == SyntaxKind.CompilationUnit;
 
             // don't reuse members if they were previously declared under a different type keyword kind
             if (this.IsIncrementalAndFactoryContextMatches)
             {
-                if (CanReuseMemberDeclaration(CurrentNodeKind))
+                if (CanReuseMemberDeclaration(CurrentNodeKind, isGlobalNonScript: isGlobal && !IsScript))
```

Comment:
feels a little weird to need to pass a global-parsing-state-flag into the nested helper.  i.e. it feels cleaner to me that CanReuseMemberDeclaration just reference IsScript within it.

Reply:
> it feels cleaner to me that CanReuseMemberDeclaration just reference IsScript within it.

Could you please elaborate? ```IsScript``` simply indicates what kind parse options we are using, it says nothing about the context.

---
In reply to: [379668721](https://github.com/dotnet/roslyn/pull/41706#discussion_r379668721) [](ancestors = 379668721)

Reply:
so it feels like you could just call `CanReuseMemberDeclaration(CurrentNodeKind, isGlobal)`.  Then, *inside* CanReuseMemberDeclaration if you care if it is `global and non-script` you can just have a local in there like `var isGlobalNonScript = isGlobal && !IsScript;`.  

In other words, It doesn't feel like teh caller has to provide this information.  But maybe i'm missing something here.

Reply:
Another way of putting it: contextual information should pass downward.  Data that is the same for the entire doc (like IsScript) doesn't need to pass down, it can just be retrieved when needed.